source/base/utilities.cc

```diff
@@ -268,6 +268,30 @@ namespace Utilities
 
 
 
+  template <int dim>
+  std::uint64_t
+  interleave(const std::array<std::uint64_t, dim> &index,
+             const int                             bits_per_dim)
+  {
+    using Integer = std::uint64_t;
+
+    AssertIndexRange(bits_per_dim * dim, 65);
+    Assert(bits_per_dim > 0, ExcMessage("bits_per_dim should be positive"));
+
+    const Integer mask = (1 << bits_per_dim) - 1;
+
+    Integer res = 0;
+    for (unsigned int i = 0; i < dim; ++i)
+      {
+        // take @bits_per_dim from each integer and shift them
+        const Integer v = (mask & index[dim - 1 - i]) << (bits_per_dim * i);
```

Comment:
any particular reason you count the indices from largest to smallest?

Reply:
because that's the representation in `inverse_Hilbert_space_filling_curve`, i.e. lexicographic order of bits: `ABCDEF` -> `ABC` + `DEF`.

Reply:
OK. Maybe state in the documentation that the first index gets the most significant bits?

Reply:
i do by saying `starting from last`, but that's not enough, you think?

Reply:
I missed this, but better be explicit.

Reply:
i updated the documentation to reflect this.